ykrt/src/compile/jitc_yk/jit_ir/mod.rs

```diff
@@ -1186,6 +1202,10 @@ impl GuardInfo {
     pub(crate) fn lives(&self) -> &Vec<InstIdx> {
         &self.lives
     }
+
+    pub(crate) fn aotlives(&self) -> &Vec<aot_ir::InstID> {
```

Comment:
What's the relationship between `lives` and `aotlives`? I guess there's a 1:1 mapping of sorts? Can we maybe assert the lengths of the two are the same? Maybe more?

Reply:
Yeah, they should have the same lenght. An assertion is a good idea.

Reply:
Maybe make it a `Vec<(..., ...)>` i.e. a `Vec` of pairs? Although perhaps that's just annoying to use in practise...

Reply:
`aotlives` will go away anyway since that's already contained in `callframes`. Same with `frames`. That's gonna be a follow up PR tho.

Reply:
Hmm, if I put asserts in here some of the unit tests fail, because they don't have any AOT IR. Since `aotlives` is going away at some point (see the FIXME here https://github.com/ykjit/yk/pull/1287/files/14e270efb2aa4213aa01d3d639e3142b388d3e68#diff-bc09a203c60315fceac958c2f894f9250466973cc0d3e51fff46e7b9efbe97a9R1179), are you ok we leave it as is for now?

Reply:
OK.